spec/controllers/attachments_controller_spec.rb

```diff
@@ -99,4 +99,44 @@
       it_behaves_like :redirected
     end
   end
+
+  describe :show do
+    let(:file) { OpenProject::Files.create_uploaded_file name: 'foobar.txt' }
+    let(:work_package) { FactoryGirl.create :work_package, project: project }
+    let(:uploader) { nil }
+
+    let(:attachment) do
+      FactoryGirl.create :attachment,
+                         container: work_package,
+                         author: user,
+                         file: file
+    end
+
+    before do
+      Attachment.mount_uploader :file, uploader
+    end
+
+    subject {
+      get :show, id: attachment.id
+    }
+
+    context 'with a local file' do
+      let(:uploader) { LocalFileUploader }
+      let(:url) { "http://test.host/attachments/#{attachment.id}/download/#{attachment.filename}"}
+
+      expect_it { to redirect_to(url) }
+    end
+
+    context 'with a remote file' do
+      let(:uploader) { FogFileUploader }
+      let(:url) do
+        host = 'https://test-bucket.s3.amazonaws.com'
+        Regexp.new "#{host}/uploads/attachment/file/#{attachment.id}/#{attachment.filename}"
+      end
+
+      it "redirects to AWS" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.